src/core/edgeNetwork/mergeLifecycleResponses.js

```diff
@@ -18,10 +19,25 @@ export default returnValues => {
   const consumerOnResponseReturnValues = returnValues.shift() || [];
   const lifecycleOnBeforeRequestReturnValues = returnValues;
 
-  return assign(
-    {},
+  return [
     ...lifecycleOnResponseReturnValues,
     ...consumerOnResponseReturnValues,
     ...lifecycleOnBeforeRequestReturnValues
-  );
+  ].reduce((accumulator, currentValue) => {
```

Comment:
This is the defect I was going to tell you about in our meeting the other day.  much improved 🏆

Reply:
i would really like this `accumulation` of results to be available as the lifecycle progresses.  I know we discussed it possibly being less performant, but might it be worth trying it out?  Changing from `promise.all` to a sequential evaluation where the responses are accumulated.  that still seems like an ideal solution for the decisioning engine propositions to be added.